packages/snaps-rpc-methods/src/permitted/revokePermissions.ts

```diff
@@ -0,0 +1,117 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  InitialPermissions,
+  RevokePermissionsParams,
+  RevokePermissionsResult,
+} from '@metamask/snaps-sdk';
+import type { Snap, SnapPermissions } from '@metamask/snaps-utils';
+import type { JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import { isPlainObject, hasProperty, assert } from '@metamask/utils';
+
+import type { MethodHooksObject } from '../utils';
+
+/**
+ * The revokePermissions hook parameters used by the `snap_revokePermissions` method.
+ *
+ */
+export type RevokePermissionsHookParams = (keyof InitialPermissions)[];
+
+export type RevokePermissionsHooks = {
+  getOriginSnap: () => Snap | undefined;
+  /**
+   * @returns Null.
+   */
+  revokePermissions: (
+    dynamicPermissions: RevokePermissionsHookParams,
+  ) => Promise<null>;
```

Comment:
```suggestion
  ) => void;
```
Revoking is synchronous and doesn't return anything

Reply:
Refactored. Same done for the `getPermissions` since it's also not async.